### PR TITLE
dotdot etc

### PR DESCRIPTION
**Pull request by KristofferC**
Adds dotdot which is exactly what you want when you compute `epsilon[Ni] * E * epsilon[Nj]` but now you have `dotdot(grad[Ni], E, grad[Nj])`.

Also cleans up some stuff, mostly in the constructors.

### CODE REVIEW

**Comment by fredrikekre** [comment]
So that would return a part of the stiffness matrix, connecting ~~dofs~~ node `i` and `j` right? So like a 2x2 matrix in 2D?

**Comment by KristofferC** [comment]
Yes

**Comment by fredrikekre** [comment]
But actually a second order tensor, and then reinterpret to get a matrix, like in the large deformation example, but there for the residual?

**Comment by KristofferC** [comment]
Can do like this:

``` jl
julia> using BlockArrays

julia> using ContMechTensors

julia> a = rand(Vec{2}); b = rand(Vec{2}); C = rand(SymmetricTensor{4,2});

julia> K = BlockArray(zeros(4,4), [2,2], [2,2])
2×2-blocked 4×4 BlockArrays.BlockArray{Float64,2,Array{Float64,2}}:
 0.0  0.0  │  0.0  0.0
 0.0  0.0  │  0.0  0.0
 ----------┼----------
 0.0  0.0  │  0.0  0.0
 0.0  0.0  │  0.0  0.0

julia> K[Block(1,2)] += dotdot(a, C, b)
2×2 Array{Float64,2}:
 1.34224  0.929681
 1.49406  1.01847

julia> K
2×2-blocked 4×4 BlockArrays.BlockArray{Float64,2,Array{Float64,2}}:
 0.0  0.0  │  1.34224  0.929681
 0.0  0.0  │  1.49406  1.01847
 ----------┼-------------------
 0.0  0.0  │  0.0      0.0
 0.0  0.0  │  0.0      0.0
```

**Comment by fredrikekre** [comment]
Thats neat!

Are the `BlockArrays` as efficient when assembling? Or would it be better to use a `PseudoBlockArray`?

**Comment by KristofferC** [comment]
Pseudo in this case.

File: src/symmetric_ops.jl [hunk 1]
```diff
@@ -218,3 +218,27 @@ function Base.eig{dim, T, M}(S::SymmetricTensor{2, dim, T, M})
     Φ = Tensor{2, dim}(ϕ)
     return Λ, Φ
 end
+
+###############################
+# Vec dot SecondOrder dot Vec #
```
Comment (fredrikekre) [inline src/symmetric_ops.jl:223]: FourthOrder :p

Comment (KristofferC) [inline src/symmetric_ops.jl:223]: argh

Comment (KristofferC) [inline src/symmetric_ops.jl:223]: i fix later, kappa

Comment (fredrikekre) [inline src/symmetric_ops.jl:223]: ░░░░▄▀▀▀▀▀█▀▄▄▄▄░░░░
░░▄▀▒▓▒▓▓▒▓▒▒▓▒▓▀▄░░
▄▀▒▒▓▒▓▒▒▓▒▓▒▓▓▒▒▓█░
█▓▒▓▒▓▒▓▓▓░░░░░░▓▓█░
█▓▓▓▓▓▒▓▒░░░░░░░░▓█░
▓▓▓▓▓▒░░░░░░░░░░░░█░
▓▓▓▓░░░░▄▄▄▄░░░▄█▄▀░
░▀▄▓░░▒▀▓▓▒▒░░█▓▒▒░░
▀▄░░░░░░░░░░░░▀▄▒▒█░
░▀░▀░░░░░▒▒▀▄▄▒▀▒▒█░
░░▀░░░░░░▒▄▄▒▄▄▄▒▒█░
░░░▀▄▄▒▒░░░░▀▀▒▒▄▀░░
░░░░░▀█▄▒▒░░░░▒▄▀░░░
░░░░░░░░▀▀█▄▄▄▄▀░░░░
░░░░░░░░░░░░░░░░░░░░

**Pull request merged**